File: 0005-longest-palindromic-substring/0005-longest-palindromic-substring.java
import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

class Solution {
    public String checkPalindrome(String s, int left, int right){
        while(left>=0 && right<s.length()){
            if(s.charAt(left)==s.charAt(right)){
                left-=1;
                right+=1;
            } else{
                break;
            }
        }
        if (left+1>right) return null;
        return s.substring(left+1,right);
    }

    public String longestPalindrome(String s) {
        int len = s.length();
        String answer=s.substring(0,1);
        for(int i=0;i<len-1;i++){
            String oddCase = checkPalindrome(s,i,i);
            String evenCase = checkPalindrome(s,i,i+1);
            if(answer.length()<evenCase.length()){
                answer=evenCase;
            }
            if(oddCase!=null && answer.length()<oddCase.length() ){
                answer=oddCase;
            }
        }
        return answer;
    }
}

File: 0005-longest-palindromic-substring/0005-longest-palindromic-substring.py
class Solution:
    def longestPalindrome(self, s: str) -> str:
        def check(left,right):
            v = ""
            while left>=0 and right<len(s) and s[left] == s[right]:
                v=s[left:right+1]
                left-=1
                right+=1
            return v
        
        answer = ''
        for i in range(len(s)):
            a = check(i,i)
            b = check(i,i+1)
            if len(answer) < len(a):
                answer = a
            if len(answer) < len(b):
                answer = b
        
        return answer